lib/ProductOpener/Products.pm

```diff
@@ -2954,6 +2955,32 @@ or "slack_CHANNEL_NAME" (B<warning> currently channel name is ignored, we post t
 
 =cut
 
+sub removeEmailValues ($product_ref) {
+
+	# Iterate over the product fields
+	foreach my $field (keys %{$product_ref->{product}}) {
+
+		if (defined $product_ref->{product}->{$field}) {
+			my $value = $product_ref->{product}->{$field};
+			my $emailRegex = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+
+			# Check if the field is multi-valued
+			if (ref($value) eq 'ARRAY') {
```

Comment:
I think it would be best to check only scalar values, and not values that are inside arrays. The _tags fields (that are arrays) may contain emails, but those are derived from other fields (e.g. brands_tags is derived from brands).

Reply:
Okay..
Experimenting with it!